### DISCUSSION TITLE
MagickNet and Acrobat Reader

### DISCUSSION THREAD

**Discussion by carbonusa** [ImageMagick/ImageMagick#2004] [General]
Hi All,
I have an issue with MagickNet and Acrobat Reader.  I convert image files to  a PDF file and then later combine that file with another PDF file using MagickNet API in C#.
The creation of this new PDF file works well but when I try to open the file using Acrobat Reader, at first the first page shows fine, but when I try to scroll I get a "Insuficient data for an image" error message.
As I scroll, I can see one more page (out of 9 pages) and also the last page which contains the images that I imported.
The file opens fine in chrome (or any other browser) and also when I use PDF Suite 2020, but not in Acrobat Reader (I need to be able to do so).
Any assistance on this matter would be greatly appreciated.

**Comment by snibgo** [comment]
I don't use Magick.NET, so probably can't help.

But your problem is too vague to answer. You should include everything needed to reproduce the problem: your code, and input files, and software versions.

**Comment by carbonusa** [comment]
This is the Magick.NET code that I use to combine the files:

public static int CombinePDFFiles(string targetPath, List<string> sourceList) 
        {
            int FileCount = 0;
            using (MagickImageCollection collection = new MagickImageCollection())
            {
                MagickReadSettings settings = new MagickReadSettings();
                settings.FrameIndex = 0; // First page
                settings.FrameCount = 100; // Number of pages


                //Add the list of files
                foreach (string sourceFile in sourceList)
                {
                    if (CheckPath(sourceFile))
                    {

                        collection.AddRange(sourceFile, settings);
                        //collection.Add(sourceFile);
                        FileCount++;
                    }
                }

                // Create combined pdf file
                if(FileCount > 0)
                    collection.Write(targetPath);
            }
            return FileCount;
        }

**Comment by dlemstra** [comment]
Next time it might be better to create this issue in the Magick.NET repository. I don't have Acrobat Reader on my machine and I really don't want to install that on my development machine. Will need to spin up a a VM and do some testing to check why this is happing but I don't think that that will happen anytime soon.
